test/location_service/location_service_test.exs

```diff
@@ -27,44 +27,43 @@ defmodule LocationServiceTest do
 
   describe "geocode/1" do
     test "can parse a response with results", %{good_response: good_response} do
-      with_mock Request, [], new: fn _ -> {:ok, good_response} end do
-        assert {:ok, body} = geocode("testing with results")
-
-        assert [
-                 %LocationService.Address{
-                   formatted: "Somewhere",
-                   longitude: -71.05566,
-                   latitude: 42.35913
-                 }
-               ] = body
-      end
+      stub(ExAws.Mock, :request, fn _ ->
+        {:ok, good_response}
+      end)
+
+      assert {:ok, body} = geocode("testing with results")
+
+      assert [
+               %LocationService.Address{
+                 formatted: "Somewhere",
+                 longitude: -71.05566,
+                 latitude: 42.35913
+               }
+             ] = body
     end
 
     test "can parse a response with no results", %{no_results: no_results} do
-      with_mock Request, [],
-        new: fn _ ->
-          {:ok, no_results}
-        end do
-        assert {:error, :zero_results} = geocode("test no results")
-      end
+      stub(ExAws.Mock, :request, fn _ ->
+        {:ok, no_results}
+      end)
+
+      assert {:error, :zero_results} = geocode("test no results")
     end
 
     test "can parse a response with no body" do
-      with_mock Request, [:passthrough],
-        new: fn _ ->
-          {:ok, %{status_code: 412}}
-        end do
-        assert {:error, :internal_error} = geocode("test no body")
-      end
+      stub(ExAws.Mock, :request, fn _ ->
+        {:ok, %{status_code: 412}}
+      end)
+
+      assert {:error, :internal_error} = geocode("test no body")
     end
 
     test "can parse a response with error" do
-      with_mock Request, [:passthrough],
-        new: fn _ ->
-          {:error, {:http_error, 500, "bad news"}}
-        end do
-        assert {:error, :internal_error} = geocode("test error")
-      end
+      stub(ExAws.Mock, :request, fn _ ->
```

Comment:
Is there a reason we're using `stub` instead of `expect` here? Stub won't verify that the function is called.

Reply:
I'd taken to using it when I don't need to verify anything specific within it, but I hadn't realized it doesn't error if it's never called -- will update